classes/utils/Location.py

```diff
@@ -0,0 +1,26 @@
+from .Position import Position
+
+class Location(Position):
+
+    def __init__(self, x = 0, y = 0, z = 0, world = None):
+        super(Location, self).__init__(x, y, z)
+        self.world = world
+    
+    def clone(self):
+        return Location(self.x, self.y, self.z, self.world)
+    
+    def to_block(self):
+        clone = super(Location, self).to_block()
+        clone.world = self.world
+        return clone
+    
+    def distance_squared(self, pos):
+        assert self.world == pos.world, 'World must be the same to calculate squared distance between two Location'
```

Comment:
How are null worlds handled? Will they pass the assert statement? Do we want that to happen?
I'd also put a comment on that the operation shouldn't be used internally, and only be used to calculate User-friendly data

Reply:
Should I test if world exist in __init__ ?

Reply:
We might, since it just is a fancier version of a worldless location